### PR TITLE
[nix.System] call EnsureNixInstalled from devbox.Open, and cleanup function API

### PR DESCRIPTION
**Pull request by savil**
## Summary

`nix.System` is widely used. So, in #1357 we wanted to change its API to just
return `string` instead of `string, error`.

To do so, we cache the result in a `nix` package variable. And populate it
during `EnsureNixInstalled`.

However, `EnsureNixInstalled` was only called from Cobra command functions. But
`devbox` as a library also now needs to call it. IMO, it should always have called
it since we do rely on `nix` being installed. This PR adds this.

It also fixes up the API to:
`System: string` and `ComputeSystem: (string, error)`.

NOTE: inside `System`, I still do a best-effort to avoid panic by calling
`ComputeSystem`. This can happen in edge-cases like some tests `generate_test/TestWriteFromTemplate`
that exercise internal functions reliant on `nix.System` without calling `devbox.Open`.

## How was it tested?

tests should pass

### CODE REVIEW

File: internal/nix/nix.go [hunk 1]
```diff
@@ -118,23 +118,21 @@ func ExperimentalFlags() []string {
...
 		}
 	}
 	return cachedSystem
 }
 
 var cachedSystem string
 
-// TODO: rename to ComputeSystem or ComputePlatform?
-func System() (string, error) {
+func ComputeSystem() (string, error) {
```
Comment (mikeland73) [inline internal/nix/nix.go:135]: ComputeSystem should only return error (we always discard the string)

**Review by mikeland73 — APPROVED** [review]

**Comment by savil** [comment]
I had to bump the test-timeout to land this. We needed this PR for correctness. But I will pursue a fix to improve the test times.
